### PR TITLE
Fix changing link on images, #803

### PR DESCRIPTION
**Pull request by anilmaurya**

### CODE REVIEW

File: src/js/core.js [hunk 1]
```diff
@@ -981,6 +981,14 @@
                                     editableElementIndex: exportedSelection.editableElementIndex
                                 }
                             );
+                            // If textNodes are not present, when changing link on images
+                            // ex: <a><img src="http://image.test.com"></a>, change fragment to currRange.startContainer
+                            // and set textNodes array to [imageElement, imageElement]
+                            if (textNodes.length === 0) {
+                                fragment = this.options.ownerDocument.createDocumentFragment();
+                                fragment.appendChild(currRange.startContainer.cloneNode(true));
+                                textNodes = [fragment.firstChild.firstChild, fragment.firstChild.lastChild];
```
Comment (nmielnik) [inline src/js/core.js]: Could `currRange.startContainer` ever be a text node?  If so, the reference to `fragment.firstChild.firstChild` and `fragment.firstChild.lastChild` could be undefined, in which case I'm not sure what the rest of the code would do.  It's also possible that this is handled ok, but I wanted to at least make sure you know what would happen in that case.

**Comment by nmielnik** [comment]
@anilmaurya Thanks a ton for the pull request and the test case, this is great!

Since you've linked to issue #803, have you verified that once an image is wrapped in a link, that you're able to actual change the url of the link?  Do you think it'd be useful to have a test case that attempts to change the href of an existing link, perhaps in `anchor-preview.spec.js` or `anchor.spec.js`?

File: src/js/core.js (hunk 1 above)
Comment (anilmaurya) [inline src/js/core.js]: If `currRange.startContainer` is text node than textNodes.length will never equal to 0 .

**Comment by anilmaurya** [comment]
@nmielnik Yes, I checked that once an image is wrapped in a link than I am able to change its url.

I updated demo/auto-link.html , to check this case. I think my test case is sufficient for this scenario, in case you think its not enough and requires more test case, Let me know.

**Comment by nmielnik** [comment]
Thanks!

**Comment by nmielnik** [comment]
@anilmaurya looks like the tests are failing on Firefox and IE11, would you be able to take a look?

**Comment by anilmaurya** [comment]
@nmielnik ok, I will check it.

**Comment by nmielnik** [comment]
### Firefox:
Firefox Failure

### IE 11:
![IE 11 Failure]
(https://assets.saucelabs.com/jobs/2912feaf41ca48949bc7dd1cadd68b96/0004screenshot.png)

**Pull request merged**